### PR TITLE
HeroImage structure

### PR DESCRIPTION
**Pull request by DamienBurkeWN**
### Testing
- Have a look at Library/Structure/HeroImage and compare with the mockups in Figma. The switch between black text/white box and white text/darkened image is via the **backgroundBox** control. Try with various mobile emulations and browsers.
- Have a look at the Page Examples/Service Landing Page Examples and verify that each look correct

### Automated test
`npm run test HeroImage`

### Live copy of Storybook to test with
[HeroImage deploy preview](https://deploy-preview-91--northants-design-system.netlify.app/?path=/story/library-slices-hero-image--hero-image-example-boxed)

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -4,38 +4,88 @@ import Button from "./library/components/Button/Button";
 import Heading from "./library/components/Heading/Heading";
 import HeadingWithIcon from "./library/components/HeadingWithIcon/HeadingWithIcon";
 import Pagination from "./library/components/Pagination/Pagination";
-import CheckboxListFilter from './library/components/CheckboxListFilter/CheckboxListFilter';
```
Comment (DamienBurkeWN) [inline src/index.ts:7]: Looks like lots of changes but mostly prettier doing its reformatting thang, only real change is adding the HeroImage slice import/export.

File: src/library/slices/HeroImage/HeroImage.types.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  /**
+   * Optional alt text for the image - ideally the image should be decorative only and not require a long description
+   */
+  imageAltText?: string;
+
+  /**
+   * Optional call to action text
+   */
+  callToActionText?: string;
```
Comment (chrisrhymes) [inline src/library/slices/HeroImage/HeroImage.types.ts:30]: Is there a way to make the callToActionText required if the callToActionURL is set?

Comment (DamienBurkeWN) [reply src/library/slices/HeroImage/HeroImage.types.ts:30]: Doesn't appear to be a straightforward way to handled that in Typescript, unless I'm missing something. I tried moving them both into a callToAction prop but then Storybook doesn't give you a nice control field to change for either - just a json field. I think the pragmatic solution would be to have some default text if the callToActionText isn't set when the URL is?

File: src/library/slices/HeroImage/HeroImage.styles.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  text-decoration: underline;
+  color: ${(props) => props.theme.theme_vars.colours.white};
+  width: 100%;
+  padding: 0;
+  margin-top: 10px;
+  display: inline-block;
+  outline: none;
+
+  &:hover,
+  &:focus {
```
Comment (chrisrhymes) [inline src/library/slices/HeroImage/HeroImage.styles.js:88]: The glow is quite subtle/hard to see. Is it worth making the underline dotted too, like other links?

Comment (DamienBurkeWN) [reply src/library/slices/HeroImage/HeroImage.styles.js:88]: 👍

**Review by chrisrhymes — APPROVED** [review]

File: src/library/pages/ServiceLandingPageExample/ServiceLandingPageExample.stories.tsx [hunk 4]
```diff
@@ -1,22 +1,46 @@
 import React from 'react';
 // also exported from '@storybook/react' if you can deal with breaking changes in 6.1
 import { Story, Meta } from '@storybook/react/types-6-0';
-
```
Comment (DamienBurkeWN) [inline src/library/pages/ServiceLandingPageExample/ServiceLandingPageExample.stories.tsx:4]: Mostly your changes to service landing pages Chris, but I've moved the data out into a storydata file too

File: src/library/pages/ServiceLandingPageExample/ServiceLandingPageExample.tsx [hunk 5]
```diff
@@ -2,233 +2,58 @@ import React from 'react';
...
-        />
+        {sections.length > 1 ? (
+          <SectionLinksMobileContents
+            sectionLinksArray={sections.map((section) => ({ id: section.sectionSlug, title: section.sectionTitle }))}
+          />
+        ) : (
+          ''
+        )}
+        {sections.map((section) => (
+          <PageStructures.SectionLinks displayTitle={sections.length > 1} {...section} key={section.sectionSlug} />
```
Comment (DamienBurkeWN) [inline src/library/pages/ServiceLandingPageExample/ServiceLandingPageExample.tsx:53]: ...titles aren't shown on sections if there's only one section

File: src/library/structure/HeroImage/HeroImage.styles.js [hunk 6]
```diff
@@ -0,0 +1,99 @@
+import styled from "styled-components";
+import Heading from "../../components/Heading/Heading";
+
+/**
+ * Hero image container with optional dark gradient for non-box mode
+ * Optimised for an image in 16:5 ratio
```
Comment (DamienBurkeWN) [inline src/library/structure/HeroImage/HeroImage.styles.js:6]: This makes the image quite high on a very big monitor like mine, so Lawrence/Richard might want this changed

File: src/library/structure/PageStructures.ts [hunk 7]
```diff
@@ -3,21 +3,21 @@ import AlertBanner from './AlertBanner/AlertBanner';
 import AlertBannerService from './AlertBannerService/AlertBannerService';
 import Breadcrumbs from './Breadcrumbs/Breadcrumbs';
 import CookieBanner from './CookieBanner/CookieBanner';
+import DisplayDate from './DisplayDate/DisplayDate';
```
Comment (DamienBurkeWN) [inline src/library/structure/PageStructures.ts:6]: Mostly just reordering for neatness in here

**Pull request merged**